scripts/sanitycheck

```diff
@@ -581,7 +581,14 @@ class BinaryHandler(Handler):
 
         start_time = time.time()
 
-        with subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, cwd=self.build_dir) as proc:
+        env = os.environ.copy()
+        env["ASAN_OPTIONS"] = "log_path=stdout:" + env.get("ASAN_OPTIONS", "")
+        # Disable memory leak detection which leads to unexpected deadlocks
+        # in posix architecture
+        if not options.enable_lsan:
+            env["ASAN_OPTIONS"] += "detect_leaks=0"
```

Comment:
Should this not be under a  `if options.enable_asan:`?
It is a bit nitpicky, but a user be enabling ASAN by other means (than with the --enable-asan option) and not expect sanitycheck to mingle with their environment behind their backs.

Reply:
fixed in #20160